lib/publishing_api_finder_publisher.rb

```diff
@@ -3,25 +3,44 @@
 require_relative "../app/presenters/finder_signup_content_item_presenter"
 
 class PublishingApiFinderPublisher
-  def initialize(metadata, schemae)
+  def initialize(metadata, schemae, log = true)
     @metadata = metadata
     @schemae = schemae
+    @log = log
   end
 
   def call
-    metadata.zip(schemae).map { |metadata, schema|
-      if metadata[:file].has_key?("content_id")
-        export_finder(metadata, schema)
-        export_signup(metadata) if metadata[:file].has_key?("signup_content_id")
+    metadata.zip(schemae).map do |metadata, schema|
+      if metadata[:file].has_key?("content_id") && !preview_only?(metadata)
+        publish metadata, schema
+      elsif preview_only?(metadata)
+        if preview_domain_or_not_production?
+          publish metadata, schema
+        else
+          puts "didn't publish #{metadata[:file]["name"]} because it is preview_only" if @log
+        end
       else
-        puts "didn't publish #{metadata[:file]["name"]} because it doesn't have a content_id"
+        puts "didn't publish #{metadata[:file]["name"]} because it doesn't have a content_id" if @log
       end
-    }
+    end
   end
 
 private
   attr_reader :schemae, :metadata
 
+  def publish metadata, schema
+    export_finder(metadata, schema)
+    export_signup(metadata) if metadata[:file].has_key?("signup_content_id")
+  end
+
+  def preview_only? metadata
+    metadata[:file]["preview_only"] == true
+  end
+
+  def preview_domain_or_not_production?
+    ENV.fetch("GOVUK_APP_DOMAIN", "")[/preview/] || !Rails.env.production?
```

Comment:
The requirement is for a preview only finder to be available in the preview environment, but not any other deployed environments. I was told the preview environment has `RAILS_ENV` set to `production`. Your suggestion of a config setting requires separate config environment files for preview, draft, and public environments. Is this the case?

This approach of identifying the preview environment using the `GOVUK_APP_DOMAIN` was recommended by @jennyd. Maybe you could discuss this further with her?